engine/execution/provider/engine.go

```diff
@@ -315,12 +315,20 @@ func (e *Engine) deliverChunkDataResponse(chunkDataPack *flow.ChunkDataPack, req
 	// sends requested chunk data pack to the requester
 	deliveryStartTime := time.Now()
 
+	nonce, err := rand.Uint64()
+	if err != nil {
+		lg.Error().
+			Err(err).
+			Msg("could not generate nonce")
+		return
+	}
+
 	response := &messages.ChunkDataResponse{
 		ChunkDataPack: *chunkDataPack,
-		Nonce:         rand.Uint64(),
+		Nonce:         nonce,
```

Comment:
it seems odd to put a random nonce in a response. I suspect this is a bug, but not in your code

Reply:
let's tag someone who may confirm this is a bug.
Btw what's the purpose of adding random nonces to requests?

Reply:
the main reason is to avoid having resubmitted requests be deduped by the network layer. They can/should also be used to match a request to a response to detect unsolicited responses.

Reply:
@kc1116 or @yhassanzadeh13, do you know if we're supposed to add a nonce to a response? Can responses be resubmitted?